unyt/tests/test_unyt_array.py

```diff
@@ -1490,8 +1490,28 @@ def op_comparison(op, inst1, inst2, compare_class):
     assert_isinstance(a.copy(), unyt_a_subclass)
     assert_isinstance(copy.deepcopy(a), unyt_a_subclass)
 
-    with pytest.raises(RuntimeError):
+
+def test_string_operations_raise_errors():
+    a = unyt_array([1, 2, 3], "g")
+    with pytest.raises(IterableUnitCoercionError):
         a + "hello"
+    with pytest.raises(IterableUnitCoercionError):
+        a * "hello"
+    with pytest.raises(IterableUnitCoercionError):
+        a ** "hello"
+    if Version(np.__version__).release[1] < 24:
```

Comment:
Just in case there's a numpy 2.x in the future
```suggestion
    if Version(np.__version__) < Version("1.24"):
```